src/models/MyAccount.ts

```diff
@@ -260,26 +296,45 @@ export default class MyAccount {
   }
 }
 
-export const MyAccountFactory = async (id: string) => {
-  client = await sierraClient()
-  const baseQuery = `patrons/${id}`
-  const checkouts = await MyAccount.fetchCheckouts(baseQuery)
-  const holds = await MyAccount.fetchHolds(baseQuery)
-  const patron = await MyAccount.fetchPatron(baseQuery)
-  const fines = await MyAccount.fetchFines(baseQuery)
-  const checkoutBibData = await MyAccount.fetchBibData(
-    checkouts.entries,
-    "item"
-  )
-  const holdBibData = await MyAccount.fetchBibData(holds.entries, "record")
-  return new MyAccount({
-    //  default to empty array to avoid hard to replicate error
-    // where entries end up undefined in buildBibData.
-    checkouts: checkouts.entries || [],
-    holds: holds.entries || [],
-    patron,
-    fines,
-    checkoutBibData: checkoutBibData.entries,
-    holdBibData: holdBibData.entries,
-  })
+export const MyAccountFactory = async (id: string, client) => {
+  const patronFetcher = new MyAccount(client, id)
+  const sierraData = await Promise.allSettled([
+    patronFetcher.getCheckouts(),
+    patronFetcher.getHolds(),
+    patronFetcher.getPatron(),
+    patronFetcher.getFines(),
+  ])
+  const [checkouts, holds, patron, fines] = sierraData.map((data) => {
+    return data.status === "fulfilled" ? data.value : []
+  }) as [Checkout[], Hold[], Patron, Fine]
+  return { checkouts, holds, patron, fines }
+}
+
+export const getPickupLocations = async () => {
```

Comment:
Where is this being used or is a stub?

Reply:
this was from account settings. its not being used because i havent decidded yet what's happening with the pickup locations fetching. but the method will be used at some point in the future. currently im just using static data for pickup locations.